File: app/src/main/java/com/vladislav/workoutassistant/ui/main/GeneralFragment.java
package com.vladislav.workoutassistant.ui.main;

import android.content.Context;

import com.vladislav.workoutassistant.ui.main.interfaces.OnFragmentListener;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class GeneralFragment extends Fragment {

    protected OnFragmentListener mFragmentListener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentListener) {
            mFragmentListener = (OnFragmentListener) context;
        } else {
            throw new IllegalStateException(context + " must implement OnFragmentListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mFragmentListener = null;
    }
}

File: app/src/main/java/com/vladislav/workoutassistant/ui/main/GeneralFragment.kt
package com.vladislav.workoutassistant.ui.main

import android.content.Context
import androidx.fragment.app.Fragment
import com.vladislav.workoutassistant.ui.main.interfaces.OnFragmentListener

open class GeneralFragment : Fragment() {

    protected var mFragmentListener: OnFragmentListener? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is OnFragmentListener) {
            mFragmentListener = context
        } else {
            throw IllegalStateException("$context must implement OnFragmentListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mFragmentListener = null
    }
}
